### PR TITLE
Metrics webapp

### PR DESCRIPTION
**Pull request by fatima-najafi**
### In this PR i implement the following:
- [x] Select API
- [x] Design
Implement  follow  [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
- Colors (select one main color for your website).
- Typography: font face, size and weight.
- Layout: composition and space between elements.
- [x] Home page
- When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
- When the user clicks (or taps) on a category item, the application navigates to the details page.
- [x] Details page
- In the details page, the webapp retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.
- [x] Testing requirements
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- Mock the access to the API, so that your tests don't send actual requests.
- [x] Technical requirements
- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- Filter the data that  retrieve from the API using a Filter stateless component.
- [x] Project documentation
- Record a video presenting the features of the project 
- A description of the project.
- A demo of the project features.
- Highlight some interesting piece of code
- [x]  Deploy the project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,12 +52,17 @@
...
+- **Fetch Data from API**
+<!-- WALKTHROUGH -->
 
-- - <!-- LIVE DEMO -->
-- ## 🚀 Live Demo <a name="live-demo"></a>
+## 🤯 Walk through video <a name="walk-through"></a>
 
-  - [Live Demo Link]Comming soon
+
+- [here](https://www.loom.com/share/b100af4d621e4a4190db3d32c3ff2026?sid=ebc6c048-9562-4cac-8009-41b658f56cd4)
```
Comment (Munsa1) [inline README.md]: - [ ] Great job creating a video presentation of your project as required, however, you are supposed to HIGHLIGHT a piece of code you found interesting or are proud of making. Kindly re-record your video and do so.

File: src/components/homepage/HomepageList.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        <input
+          className="input"
+          type="text"
+          onChange={(e) => setSearch(e.target.value)}
+          placeholder="Search Company"
+          value={search}
+        />
+      </div>
+
+      <div className="container">
```
Comment (Munsa1) [inline src/components/homepage/HomepageList.js:51]: - [ ] Great job on your project it works fine. However, your design could use some improvements so as to match the given design

Given design
image

Your design
image

- Consider removing the white space around your cards and have them each fill up 50% of the mobile view so as to match the given design. Then you can give alternating cards the same color of grey and the other alternating cards should have a slightly lighter variation of the grey color
image
Notice how the cards have a color pattern where the first card is slightly lighter as compared to the next card.
- Also notice how there is no spacing between the cards in the given design. Kindly implement that as well.
- Also consider aligning your text on each card to the bottom right as is in the given design and the image to the top left corner.

- [ ] Also consider using the provided font family for your design. You are using `Poppins` whilst the given one is `gill sans` and `Lato`. This applies to all text in your app.
image

File: src/components/details/Detailspage.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          {companyDetails.symbol}
+          )
+        </p>
+        <h1>{companyDetails.companyName}</h1>
+        <table className="table">
+          <tbody>
+            <tr>
+              <th scope="col">CEO</th>
+              <td>{companyDetails.ceo}</td>
+            </tr>
```
Comment (Munsa1) [inline src/components/details/Detailspage.js]: - [ ] The same applies to the details page, kindly ensure you follow the given design.

Your design
image

Given design
image

- Kindly ensure your design fills up the whole view of the mobile design, i.e. there should be no white space around your design
- The font family used should also be consistent with the given design
- The image in the details page should be aligned to the left and the text to the right as is in the given design.

File: src/components/navbar/Navbar.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          src={back}
+          alt="back"
+        />
+      </div>
+      <h2 className="heading">Companies List</h2>
+      <div className="position">
+        <img className="btn" src={mic} alt="mic" />
+        <img className="btn" src={setting} alt="setting" />
+      </div>
+    </div>
```
Comment (Munsa1) [inline src/components/navbar/Navbar.js]: - [ ] One of the requirements is that you write Integration tests for your application, I would advise to write for the navigation bar. Check out a similar example below

```
Sample navigation bar
// NavigationBar.js
import React from 'react';
import { Link } from 'react-router-dom';

const NavigationBar = () => {
  return (
    <nav>
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/about">About</Link></li>
        <li><Link to="/contact">Contact</Link></li>
      </ul>
    </nav>
  );
};

export default NavigationBar;
```
```
Test example

// NavigationBar.test.js
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NavigationBar from './NavigationBar';

test('navigation links work correctly', () => {
  const { getByText, history } = render(
    <MemoryRouter>
      <NavigationBar />
    </MemoryRouter>
  );

  // Initially, we're on the Home page
  expect(getByText('Home')).toBeInTheDocument();
  expect(getByText('About')).toBeInTheDocument();
  expect(getByText('Contact')).toBeInTheDocument();

  // Click on the About link
  fireEvent.click(getByText('About'));
  expect(history.location.pathname).toBe('/about');

  // Click on the Contact link
  fireEvent.click(getByText('Contact'));
  expect(history.location.pathname).toBe('/contact');

  // Click on the Home link
  fireEvent.click(getByText('Home'));
  expect(history.location.pathname).toBe('/');
});
```

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @fatima-najafi, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 5]
```diff
@@ -51,12 +52,17 @@
...
+<!-- WALKTHROUGH -->
 
-- - <!-- LIVE DEMO -->
-- ## 🚀 Live Demo <a name="live-demo"></a>
+## 🤯 Walk through video <a name="walk-through"></a>
 
-  - [Live Demo Link]Comming soon
+
+- [here](https://www.loom.com/share/03d5c7e71a604a17be31b44890439338?sid=3299bd83-b4a9-41c4-be73-917804948ea7)
+ <!-- LIVE DEMO -->
```
Comment (usorfaitheloho) [inline README.md]: As suggested by the previous reviewer, kindly HIGHLIGHT a piece of code you found interesting or are proud of making in your video, you could redo the video to state this point 😄

File: src/components/homepage/HomepageList.js (hunk 2 above)
Comment (usorfaitheloho) [reply src/components/homepage/HomepageList.js:51]: You did well fixing some of the problems identified by the previous reviewer, kindly  fix this one `Also consider aligning your text on each card to the bottom right as is in the given design and the image to the top left corner.`

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
Hi @fatima-najafi  :thumbsup:  ,

This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are still some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Fixed some changes suggested by the previous reviewer ✔️ 
<br/>
<br/>
<br/>

 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @fatima-najafi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**